pinot-common/src/main/java/com/linkedin/pinot/common/data/MetricFieldSpec.java

```diff
@@ -42,4 +70,30 @@ public FieldType getFieldType() {
   public void setSingleValueField(boolean isSingleValueField) {
     Preconditions.checkArgument(isSingleValueField, "Unsupported multi-value for metric field.");
   }
+
+  public boolean isDerivedMetricType(DerivedMetricType type) {
+    return derivedMetricType == type;
+  }
+
+  public boolean isDerivedMetric() {
+    return derivedMetricType != null;
+  }
+
+  public void setDerivedMetricType(DerivedMetricType derivedMetricType) {
```

Comment:
Add a getter method for derivedMetricType: public DerivedMetricType isDerivedMetricType()